pkg/resources/task.go

```diff
@@ -344,156 +258,180 @@ func ReadTask(d *schema.ResourceData, meta interface{}) error {
 
 // CreateTask implements schema.CreateFunc.
 func CreateTask(d *schema.ResourceData, meta interface{}) error {
-	var err error
 	db := meta.(*sql.DB)
-	database := d.Get("database").(string)
-	schema := d.Get("schema").(string)
+	client := sdk.NewClientFromDB(db)
+	ctx := context.Background()
+
+	databaseName := d.Get("database").(string)
+	schemaName := d.Get("schema").(string)
 	name := d.Get("name").(string)
-	sql := d.Get("sql_statement").(string)
-	enabled := d.Get("enabled").(bool)
 
-	builder := snowflake.NewTaskBuilder(name, database, schema)
-	builder.WithStatement(sql)
+	sqlStatement := d.Get("sql_statement").(string)
+
+	taskId := sdk.NewSchemaObjectIdentifier(databaseName, schemaName, name)
+	createRequest := sdk.NewCreateTaskRequest(taskId, sqlStatement)
 
 	// Set optionals
 	if v, ok := d.GetOk("warehouse"); ok {
-		builder.WithWarehouse(v.(string))
+		warehouseId := sdk.NewAccountObjectIdentifier(v.(string))
+		createRequest.WithWarehouse(sdk.NewCreateTaskWarehouseRequest().WithWarehouse(&warehouseId))
 	}
 
 	if v, ok := d.GetOk("user_task_managed_initial_warehouse_size"); ok {
-		builder.WithInitialWarehouseSize(v.(string))
+		size, err := sdk.ToWarehouseSize(v.(string))
+		if err != nil {
+			return err
+		}
+		createRequest.WithWarehouse(sdk.NewCreateTaskWarehouseRequest().WithUserTaskManagedInitialWarehouseSize(&size))
 	}
 
 	if v, ok := d.GetOk("schedule"); ok {
-		builder.WithSchedule(v.(string))
+		createRequest.WithSchedule(sdk.Pointer(v.(string)))
```

Comment:
this could be sdk.String()

Reply:
Used `sdk.String`, `sdk.Int` and `sdk.Bool` instead of `sdk.Pointer` throughout this file.